src/components/api.js

```diff
@@ -1,50 +1,63 @@
 import { state } from '../index';
 
+const BASE_URL = 'https://realtor.p.rapidapi.com';
+const LOCATIONS_AUTOCOMPLETE_URL = BASE_URL + '/locations/auto-complete';
+const PROPERTIES_FOR_SALE_URL = BASE_URL + '/properties/v2/list-for-sale';
+const API_KEY = '897733c2d3msh13b53444c4f53c4p152d7fjsn04123a994089';
+
+const propertiesBuffer = new Map();
+
 export async function getSuggestions(location = 'a') {
-  if (!location) {
-    location = 'a';
-  }
+  if (!location) location = 'a';
+
   try {
-    const res = await fetch(
-      encodeURI(
-        `https://realtor.p.rapidapi.com/locations/auto-complete?input=${location}`
-      ),
-      {
-        method: 'GET',
-        headers: {
-          'x-rapidapi-host': 'realtor.p.rapidapi.com',
-          'x-rapidapi-key':
-            '912915357fmshd8b3c8fc8fcaa14p122a7ejsnb050b17596f1',
-        },
-      }
-    );
+    const URL_PARAMS = encodeURI(`?input=${location}`);
+    const fetchMethod = 'GET';
+    const xRapidapiHost = 'realtor.p.rapidapi.com';
+
+    const res = await fetch(BASE_URL + LOCATIONS_AUTOCOMPLETE_URL + URL_PARAMS, {
```

Comment:
const BASE_URL = 'https://realtor.p.rapidapi.com';
const LOCATIONS_AUTOCOMPLETE_URL = BASE_URL + '/locations/auto-complete';
BASE_URL + LOCATIONS_AUTOCOMPLETE_URL = 'https://realtor.p.rapidapi.com' + 'https://realtor.p.rapidapi.com' + '/locations/auto-complete'

Are you sure it's correct url?